### PR TITLE
Migrate pub_server's startUpload.

### PR DESCRIPTION
**Pull request by isoos**
- #3343
- tested with 1.24.3 and 2.8.0-dev.17.0 too

### CODE REVIEW

File: pkg/client_data/lib/package_api.dart [hunk 1]
```diff
@@ -3,9 +3,33 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'package:json_annotation/json_annotation.dart';
+import 'package:meta/meta.dart';
 
 part 'package_api.g.dart';
 
+/// Information obtained when starting an asynchronous upload.
+@JsonSerializable()
+class AsyncUploadInfo {
```
Comment (jonasfj) [inline pkg/client_data/lib/package_api.dart:12]: I think we can just call it `UploadInfo` we have no sync-flow.

File: pkg/client_data/lib/package_api.dart [hunk 2]
```diff
@@ -3,9 +3,33 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'package:json_annotation/json_annotation.dart';
+import 'package:meta/meta.dart';
 
 part 'package_api.g.dart';
 
+/// Information obtained when starting an asynchronous upload.
```
Comment (jonasfj) [inline pkg/client_data/lib/package_api.dart:10]: Information for uploading a package.

File: pkg/client_data/lib/package_api.dart (hunk 1 above)
Comment (jonasfj) [reply pkg/client_data/lib/package_api.dart:12]: Or we can call it `UploadUrl` or `UploadUrlInfo`

File: app/lib/package/backend.dart [hunk 3]
```diff
@@ -433,7 +433,7 @@ class GCloudPackageRepository extends pub_server.PackageRepository {
   }
 
   @override
-  Future<pub_server.AsyncUploadInfo> startAsyncUpload(Uri redirectUrl) async {
+  Future<api.AsyncUploadInfo> startAsyncUpload(Uri redirectUrl) async {
```
Comment (jonasfj) [inline app/lib/package/backend.dart:436]: Let's rename this method and call it something else.

Comment (isoos) [reply app/lib/package/backend.dart]: Maybe `startUpload`?

Comment (isoos) [reply app/lib/package/backend.dart]: (merging this, we can do the rename later if `startUpload` is not the best one...)

Comment (jonasfj) [reply app/lib/package/backend.dart]: Or `getUploadUrlInfo` or similar...

**Pull request merged**